internal/upgrader/upgrader_test.go

```diff
@@ -232,6 +232,98 @@ var _ = Describe("Upgrade", func() {
+					Expect(upgradeErr2).To(Equal(expectedErr))
+				}
+				if expectedLog == "" {
+					Expect(upgradeLog).To(Equal(""))
+					Expect(upgradeLog2).To(Equal(""))
+				} else {
+					Expect(upgradeLog).To(MatchRegexp(expectedLog))
+					Expect(upgradeLog2).To(MatchRegexp(expectedLog))
+				}
+			},
+			Entry("no plans defined",
+				fmt.Errorf("no service plans available for broker: fake-broker-name"), "",
+				[]ccapi.Plan{},
+				[]ccapi.ServiceInstance{},
+			),
+			Entry("no instances defined",
+				nil, "no instances available to upgrade",
+				[]ccapi.Plan{fakePlan},
+				[]ccapi.ServiceInstance{},
+			),
+			Entry("all instances up to date",
+				nil, "no instances available to upgrade",
+				[]ccapi.Plan{fakePlan},
+				[]ccapi.ServiceInstance{{UpgradeAvailable: false}},
+			),
+			Entry("upgradable instances",
+				fmt.Errorf("check up-to-date failed: found 1 instances which are not up-to-date"),
+				`(\s|.)*`, // we don't care about the logs in this test
+				[]ccapi.Plan{fakePlan},
+				[]ccapi.ServiceInstance{{UpgradeAvailable: true}},
+			),
+			Entry("upgradable instances detailed logs",
+				fmt.Errorf("check up-to-date failed: found 1 instances which are not up-to-date"),
+				getExpectedLogForInstance(getFakeInstanceDetailed()),
+				[]ccapi.Plan{{GUID: getFakeInstanceDetailed().PlanGUID, MaintenanceInfoVersion: getFakeInstanceDetailed().PlanMaintenanceInfoVersion}},
+				[]ccapi.ServiceInstance{getFakeInstanceDetailed()},
+			),
+			Entry("upgradeable instances whose creation failed",
+				nil, "no instances available to upgrade",
+				[]ccapi.Plan{{GUID: fakeInstance1.PlanGUID}},
+				[]ccapi.ServiceInstance{{UpgradeAvailable: true, LastOperation: ccapi.LastOperation{Type: "create", State: "failed"}}},
+			),
+			/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+			// WARNING:
+			// We may want to check whether the following scenarios are possible and if they do, we may want to prevent them from happening
+			/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+			Entry("upgradeable instances whose creation failed and are in a possible invalid state?",
+				fmt.Errorf("check up-to-date failed: found 1 instances which are not up-to-date"),
+				`(\s|.)*`, // we don't care about the logs in this test
+				[]ccapi.Plan{{GUID: fakeInstance1.PlanGUID}},
+				[]ccapi.ServiceInstance{{UpgradeAvailable: true, LastOperation: ccapi.LastOperation{Type: "destroy", State: "failed"}}},
+			),
+			Entry("unlikely scenario in which the api returns instances which doesn't correspond to the queried plan",
+				fmt.Errorf("check up-to-date failed: found 1 instances which are not up-to-date"),
+				`(\s|.)*`, // we don't care about the logs in this test
+				[]ccapi.Plan{{GUID: "guid-for-unlikely-scenario"}},
+				[]ccapi.ServiceInstance{getFakeInstanceDetailed()},
+			),
```

Comment:
```suggestion
```

This test is unrealistic and doesn't provide real value.
This test is faking the result of two API calls, returning invalid data. So this test assumes API is behaving abnormally, which is a very unlikely scenario and out of our control.

However, we are already analysing a more realistic scenario:

When a Plan is deleted but it contains active instances Cloudfoundry disables the plan instead of deleting it
Disabled instances may contain a MaintenanceInfoVersion which is not representative of the current version of the Tile being installed (apparently, MaintenaceInfoVersion for disabled plans doesn't get updated)
Therefore, the mechanism for checking if instances need upgrading will determine that there are no upgrades available
After we finish analysing this corner-case we will make any required changes. For now, let's remove this useless test.